backend/app/audio.py

```diff
@@ -1,28 +1,28 @@
 """audio processing, etc"""
 from pydub import AudioSegment
+import pandas as pd
+import numpy as np
+import soundfile as sf
 
 
 class Audio:
     silence_threshold = -50.0
-    chunk_size = 10
+    threshold_value = 0.0007  # Tweak the value of threshold to get the aggressive trimming
 
     @staticmethod
-    def _detect_leading_silence(sound: AudioSegment) -> int:
-        trim_ms = 0
-        assert Audio.chunk_size > 0  # to avoid infinite loop
-        while sound[trim_ms:trim_ms + Audio.chunk_size].dBFS \
-                < Audio.silence_threshold and trim_ms < len(sound):
-            trim_ms += Audio.chunk_size
+    def _detect_leading_silence(sound: bytearray, sample_rate: int) -> list:
```

Comment:
The name of this method should probably be changed to reflect the new nature of it. It is no longer "detecting leading silence" it is generating a mask.

Reply:
yes I am working on it